app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,63 @@
+class FriendshipsController < ApplicationController
+  @is_unfriend = false
```

Comment:
what's the purpose of this instance variable in global controller scope?

Reply:
using it in destroy method to differentiate between request for 'unfriend' or 'cancel request'

Reply:
Think of something more conventional. This does not seem to be good strategy to use global instance variable in controller.